### DISCUSSION TITLE
Sea-ORM migration crate panic on cli Migration::up

### DISCUSSION THREAD

**Discussion by hyperliskdev** [SeaQL/sea-orm#1079] [Q&A] [answered]
I am having some trouble with sea-orm-migration.

I cant seem to get my migration to start up. If anyone has any tips, I keep getting this same panic as;

``` thread 'main' panicked at 'not implemented', /home/****/.cargo/registry/src/github.com-*/sea-query-0.26.3/src/backend/postgres/table.rs:239:26 ```

I have a backtrace if needed. I am a little lost right now. Any guidance is appreciated.

**Comment by hyperliskdev** [comment]
> I am having some trouble with sea-orm-migration.
> 
> I cant seem to get my migration to start up. If anyone has any tips, I keep getting this same panic as;
> 
> `thread 'main' panicked at 'not implemented', /home/****/.cargo/registry/src/github.com-*/sea-query-0.26.3/src/backend/postgres/table.rs:239:26`
> 
> I have a backtrace if needed. I am a little lost right now. Any guidance is appreciated.

Specifically this

``` 
impl PostgresQueryBuilder {
    fn prepare_column_type_check_auto_increment(
        &self,
        column_def: &ColumnDef,
        sql: &mut SqlWriter,
    ) {
        if let Some(column_type) = &column_def.types {
            write!(sql, " ").unwrap();
            let is_auto_increment = column_def
                .spec
                .iter()
                .position(|s| matches!(s, ColumnSpec::AutoIncrement));
            if is_auto_increment.is_some() {
                match &column_type {
                    ColumnType::SmallInteger(_) => write!(sql, "smallserial").unwrap(),
                    ColumnType::Integer(_) => write!(sql, "serial").unwrap(),
                    ColumnType::BigInteger(_) => write!(sql, "bigserial").unwrap(),
                    _ => unimplemented!(), **UNIMPLEMENTED SOMETHING?**
                }
            } else {
                self.prepare_column_type(column_type, sql);
            }
        }
    }
}
```

Why would there be this unimplemented!() macro?

**Comment by billy1624** [comment, marked as answer]
Hey @hyperliskdev, can you show me your migration script? Seems your auto incremented column isn't of integer types?

**Reply by hyperliskdev** [reply]
Oh no that's embarrassing :( thank you very much.

**Reply by billy1624** [reply]
No worries!! :D

**Comment by tyt2y3** [comment]
Umm... is there anyway we can provide more context in the error message ? @billy1624
